src/test/java/seedu/address/logic/commands/BuyCommandTest.java

```diff
@@ -108,14 +114,16 @@ public void execute_buyOverflowInventory_throwsCommandException() {
 
     @Test
     public void execute_validTransaction_callsModelCommit() throws CommandException {
-        ModelStubCommit modelStub = new ModelStubCommit();
+        ModelStubWithExistingGood modelStub = new ModelStubWithExistingGood();
         new BuyCommand(boughtGood).execute(modelStub);
```

Comment:
Why change the stub class?

Reply:
Now BuyCommand requires a supplier list within the ModelStub to check against. 

Wanted to keep your ModelStubCommit for only testing of commits and not pollute it with unnecessary member fields, since your ModelStubCommit is used in the testing of every command.

Reply:
ModelStubCommit should have a supplier list, since it inherits from ModelManager

Reply:
ModelStubCommit that inherits from ModelManager has a supplier list, but it's not populated with the test supplier information. 

Having that be initialized and making a stub method that returns that test supplier list means implementing those methods in your ModelStubCommit, and pulling in the dummy test suppliers already created in ModelStubWithExistingGood.

Do you still want me to carry out that refactor?

Reply:
I guess we can leave it first